src/gt4py/next/ffront/lowering_utils.py

```diff
@@ -113,34 +114,36 @@ def process_elements(
             are not used and thus not relevant.
     """
     if isinstance(objs, itir.Expr):
-        objs = [objs]
+        objs = (objs,)
 
-    _current_el_exprs = [
-        im.ref(f"__val_{eve_utils.content_hash(obj)}") for i, obj in enumerate(objs)
-    ]
-    body = _process_elements_impl(process_func, _current_el_exprs, current_el_type)
-
-    return im.let(*((f"__val_{eve_utils.content_hash(obj)}", obj) for i, obj in enumerate(objs)))(  # type: ignore[arg-type]  # mypy not smart enough
-        body
+    let_ids = tuple(f"__val_{eve_utils.content_hash(obj)}" for obj in objs)
+    body = _process_elements_impl(
+        process_func, tuple(im.ref(let_id) for let_id in let_ids), current_el_type
     )
 
+    return im.let(*(zip(let_ids, objs, strict=True)))(body)
+
 
 T = TypeVar("T", bound=itir.Expr, covariant=True)
 
 
 def _process_elements_impl(
-    process_func: Callable[..., itir.Expr], _current_el_exprs: list[T], current_el_type: ts.TypeSpec
+    process_func: Callable[..., itir.Expr],
+    _current_el_exprs: Iterable[T],
+    current_el_type: ts.TypeSpec,
 ) -> itir.Expr:
     if isinstance(current_el_type, ts.TupleType):
         result = im.make_tuple(
-            *[
+            *(
```

Comment:
The reason I always use lists by default instead of generators is because it is easier to evaluate in the debugger and a list comprehension is shorter then a tuple comprehension (like below).

Reply:
I agree with the interpretation. At some point I just switched because the aforementioned reasons bugged me more.